addons/controllers/kapp-controller/kappcontrollerconfig_types.go

```diff
@@ -93,6 +93,7 @@ func mapKappControllerConfigSpec(cluster *clusterapiv1beta1.Cluster, config *run
 	configSpec.KappController.Deployment.PriorityClassName = config.Spec.KappController.Deployment.PriorityClassName
 	configSpec.KappController.Deployment.Concurrency = config.Spec.KappController.Deployment.Concurrency
 	configSpec.KappController.Deployment.Tolerations = config.Spec.KappController.Deployment.Tolerations
+	configSpec.KappController.Deployment.Tolerations = append(configSpec.KappController.Deployment.Tolerations, map[string]string{"effect": "NoSchedule", "key": "node-role.kubernetes.io/control-plane"})
```

Comment:
I think you want to append if not exists.

Reply:
Done.